### PR TITLE
Generate bearing-transition style property.

### PR DESCRIPTION
**Pull request by yunikkk**
### Summary of changes

Generated property bearing-transition

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [ ] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [x] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-kotlin-binary-compatibility` CI will fail.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [ ] Add an entry inside this element for inclusion in the `mapbox-maps-android` changelog: `<changelog></changelog>`.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

**Comment by pengdev** [comment]
@yunikkk please update the api files and link the generator change from internal repo.

**Comment by ank27** [comment]
@yunikkk can you link internal pr as well, since these are generated files.

**Comment by pengdev** [comment]
@yunikkk the verify-code bot has failed, could you fix it and manually trigger the `ci/circleci: build/start-run-style-test`?

**Comment by yunikkk** [comment]
> @yunikkk the verify-code bot has failed, could you fix it and manually trigger the `ci/circleci: build/start-run-style-test`?

Yes, will get back to this one after the release

**Comment by pengdev** [comment]
This PR in theory breaks the semver, as we added new methods to `LocationIndicatorLayerDsl` interface. However we could treat it as a bug fix as this is a missing API(supported but not exposed). We need to revisit the generated layer DSLs to use builder patterns instead of interfaces in the next major release.

**Comment by yunikkk** [comment]
> This PR in theory breaks the semver, as we added new methods to `LocationIndicatorLayerDsl` interface. However we could treat it as a bug fix as this is a missing API(supported but not exposed). We need to revisit the generated layer DSLs to use builder patterns instead of interfaces in the next major release.

Right, fully agree.

File: extension-style/src/test/java/com/mapbox/maps/extension/style/layers/generated/LocationIndicatorLayerTest.kt [hunk 1]
```diff
@@ -724,6 +724,46 @@ class LocationIndicatorLayerTest {
...
+  @Test
+  fun bearingTransitionSetDsl() {
+    val layer = locationIndicatorLayer("id") {}
+    layer.bindTo(style)
+    layer.bearingTransition {
+      duration(100)
+      delay(200)
+    }
+    verify { style.setStyleLayerProperty("id", "bearing-transition", capture(valueSlot)) }
+    assertEquals(valueSlot.captured.toString(), "{duration=100, delay=200}")
```
Comment (Chaoba) [inline extension-style/src/test/java/com/mapbox/maps/extension/style/layers/generated/LocationIndicatorLayerTest.kt:764]: nit, the expect value should be the first param

**Review by Chaoba — APPROVED** [review]

File: extension-style/src/test/java/com/mapbox/maps/extension/style/layers/generated/LocationIndicatorLayerTest.kt (hunk 1 above)
Comment (yunikkk) [reply extension-style/src/test/java/com/mapbox/maps/extension/style/layers/generated/LocationIndicatorLayerTest.kt:764]: Hmm but it's generated test...

**Pull request merged**